src/types_unstable.ts

```diff
@@ -0,0 +1,54 @@
+export type * from './aggregate.js';
+export type * from './axis.js';
+export type * from './bin.js';
+export type * from './channel.js';
+export type * from './channeldef.js';
+export type * from './compositemark/base.js';
+export type * from './compositemark/boxplot.js';
+export type * from './compositemark/common.js';
+export type * from './compositemark/errorband.js';
+export type * from './compositemark/errorbar.js';
+export type * from './compositemark/index.js';
+export type * from './data.js';
+export type * from './datetime.js';
+export type * from './encoding.js';
+export type * from './expr.js';
+export type * from './guide.js';
+export type * from './header.js';
+export type * from './impute.js';
+export type * from './index.js';
+export type * from './legend.js';
+export type * from './log/index.js';
+export type * from './log/message.js';
+export type * from './logical.js';
+export type * from './mark.js';
+export type * from './normalize/base.js';
+export type * from './normalize/core.js';
+export type * from './normalize/index.js';
+export type * from './normalize/pathoverlay.js';
+export type * from './normalize/repeater.js';
+export type * from './normalize/ruleforrangedline.js';
+export type * from './normalize/selectioncompat.js';
+export type * from './normalize/toplevelselection.js';
+export type * from './parameter.js';
+export type * from './predicate.js';
+export type * from './projection.js';
+export type * from './resolve.js';
+export type * from './scale.js';
+export type * from './selection.js';
+export type * from './sort.js';
+export type * from './spec/base.js';
+export type * from './spec/concat.js';
+export type * from './spec/facet.js';
+export type * from './spec/index.js';
```

Comment:
To maintain parity with `5.23.0`, I think only 2 types have to be exported

- `Config`
- `TopLevelSpec`

https://github.com/vega/vega-lite/blob/cbad08391897b835e0a27956faef7a75afdae4c9/src/index.ts#L5-L7

I recognize we're adding more in order to address the fact that some teams were importing directly from the src folder before and bypassing the index.ts route.

While we could export the rest, if we're concerned about instability, we'll have an easier time if we err on the side of exporting fewer.

If we have multiple types that have the same name from any of these files, they may end up colliding in this file. It's not a huge deal while this file is marked unstable, but it's something we'll want to guard and have linting for before we make this API stable.

Reply:
We do keep the exports in index.ts, though. The unstable type exports are in addition.

Reply:
Yeah, something like `find . -type f -exec echo {} \;  | sed 's/\(.*\)/export type \* from \"\1\"/'` would be nice but it's too generic and creates conflicts.